protected/humhub/modules/user/stream/filters/IncludeAllContributionsFilter.php

```diff
@@ -50,27 +50,34 @@ public function apply()
         $this->query->leftJoin('space', 'contentcontainer.pk=space.id AND contentcontainer.class=:spaceClass', [':spaceClass' => Space::class]);
         $this->query->leftJoin('user cuser', 'contentcontainer.pk=cuser.id AND contentcontainer.class=:userClass', [':userClass' => User::class]);
 
-        $this->query->leftJoin('space_membership',
-            'contentcontainer.pk=space_membership.space_id AND contentcontainer.class=:spaceClass AND space_membership.user_id=:userId',
-            [':userId' => $queryUser->id, ':spaceClass' => Space::class]
-        );
-
         $this->query->andWhere([
             'OR',
             ['content.created_by' => $this->container->id],
             ['content.contentcontainer_id' => $this->container->contentcontainer_id]
         ]);
 
+        if ($queryUser->canViewAllContent()) {
+            // Don't restrict if user can view all content:
+            $conditionSpaceMembershipRestriction = '';
+            $conditionUserPrivateRestriction = '';
+        } else {
+            // User must be a space's member OR Space and Content are public
+            $this->query->leftJoin('space_membership',
```

Comment:
Is this the same condition as above? Maybe we can refactor it somehow to prevent redundance of a rather complex code section.

Reply:
yes, I too thought about how to refactor them, ok I will review this again.

Reply:
If we do not find a satisfying solution just add a `TODO: Refactor to unify with ActiveQueryContent::readable()`

Reply:
I couldn't refactor, as I guess we should replace `$this->_query = Content::find();` in https://github.com/humhub/humhub/blob/enh/48-improve-admin-can-view-all-feature/protected/humhub/modules/stream/models/StreamQuery.php#L193
`\humhub\modules\stream\models\StreamQuery->init()` to smth like this `$this->_query = ContentActiveRecord::find();` but I tried this and it doesn't work.

Reply:
ok, no big issue just doesn't look very clean maybe we can refactor it when implementing the `ActiveQueryContent::visible()` enhancement